dashboard/pages/api/metric.ts

```diff
@@ -17,17 +17,148 @@
 import { Metrics, MetricsSample } from "../../components/metrics"
 import api from "./api"
 
+export const INTERVAL = 5000
 export interface BackPressuresMetrics {
   outputBufferBlockingDuration: Metrics[]
 }
 
+// Get back pressure from meta node -> prometheus
 export async function getActorBackPressures() {
   const res: BackPressuresMetrics = await api.get(
     "/metrics/actor/back_pressures"
   )
   return res
 }
 
+export interface BackPressureInfo {
+  actorId: number
+  fragmentId: number
+  downstreamFragmentId: number
+  value: number
+}
+
+export interface BackPressureRateInfo {
+  actorId: number
+  fragmentId: number
+  downstreamFragmentId: number
+  backPressureRate: number
+}
+
+function convertToMapAndAgg(
+  backPressures: BackPressureInfo[]
+): Map<string, number> {
+  // FragmentId-downstreamFragmentId, total value
+  const mapValue = new Map<string, number>()
+  // FragmentId-downstreamFragmentId, total count
+  const mapNumber = new Map<string, number>()
+  // FragmentId-downstreamFragmentId, average value
+  const map = new Map<string, number>()
+  for (const item of backPressures) {
+    const key = `${item.fragmentId}-${item.downstreamFragmentId}`
+    if (mapValue.has(key) && mapNumber.has(key)) {
+      // add || tp avoid NaN and pass check
+      mapValue.set(key, (mapValue.get(key) || 0) + item.value)
+      mapNumber.set(key, (mapNumber.get(key) || 0) + 1)
+    } else {
+      mapValue.set(key, item.value)
+      mapNumber.set(key, 1)
+    }
+  }
+
+  for (const [key, value] of mapValue) {
+    // add || 1 to avoid NaN and pass check
+    map.set(key, value / (mapNumber.get(key) || 1))
```

Comment:
Use `!` to tell TS compiler that the value must be non-null.

```suggestion
    map.set(key, value / mapNumber.get(key)!)
```